docs/introduction.rst

```diff
@@ -0,0 +1,111 @@
+Introduction
+============
+
+The XAITK-Saliency toolkit implements a class of XAI algorithms known
+as `saliency algorithms`. The basic application pipeline is shown in Figure 1:
+
+.. figure:: figures/intro-fig-01.png
+
+*Figure 1: A basic AI pipeline.*
+
+In this scenario, an AI algorithm operates on an input (text, image,
+etc.) to produce some sort of output. Saliency algorithms build on
+this as shown in Figure 2:
+
+.. figure:: figures/intro-fig-02.png
+
+*Figure 2: The AI pipeline augmented with a saliency algorithm.*
+
+At a high level, saliency maps are typically a copy of the input with
+regions highlighted to indicate areas that are somehow significant to
+the AI. Figure 3 shows sample saliency maps for text and images.
+
+.. figure:: figures/intro-fig-03.png
+
+*Figure 3: Sample saliency maps for (left) text (from* `Tuckey et al.
+<https://arxiv.org/abs/1907.05664>`_) *and (right) images (from* `Dong et
+al. <https://openaccess.thecvf.com/content_CVPRW_2019/html/Explainable_AI/Dong_Explainability_for_Content-Based_Image_Retrieval_CVPRW_2019_paper.html>`_
+*)*
+
+.. note:: The XAITK-Saliency toolkit focuses on providing saliency
+          maps for images.
+
+Image Saliency Maps: An Intuitive Introduction
+----------------------------------------
+
+Figure 4 shows a deep learning pipeline for recognizing objects in
+images; pixels in the image (in green) are processed by the pipeline
+to produce the output (in orange). Here, the system has been trained
+to recognize 1000 object types. Its output is a list of 1000 numbers,
+one for each object type; each number is between 0 and 1, representing
+the system's estimate of whether or not that particular object is in
+the image.
+
+.. figure:: figures/cnn-tagged-scaled.png
+
+*Figure 4: Typical object recognition CNN architecture; the image
+(green, left) is processed left to right through the CNN to produce an
+output vector (orange, right). Diagram from* `Krizhevsky et
+al. <https://proceedings.neurips.cc/paper/4824-imagenet-classification-with-deep-convolutional-neural-networks.pdf>`_
+
+
+This operation is "all or nothing" at both ends: the entire image must
+be processed, and the the entire vector must be output. There is no
```

Comment:
The motivation for this paragraph is a bit unclear- is this mainly harping on the "black box" nature of these types of systems?

Reply:
I assume we don't want to use "black-box" here because that term is introduced later. Maybe we want to more phrase this in terms of different target audiences not being able to understand the "why" of the results from just the input, the outputs and the raw algorithm, like the following questions you pose in bullet points.

Reply:
Yes, although not so much black-box / white-box but rather the monolithicity of a CNN independent of XAI.

Reply:
@brianhhu do we have any more specific feedback for this or can this be the way it is for this PR?

Reply:
I think this is fine for now, we can always choose to revisit this after the other sections are completed.